src/payment/checkout/payment-form/flex-microform/FlexMicroformField.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faLock, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+import ReactTooltip from 'react-tooltip';
+
+class FlexMicroformField extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            field: null
+        }
+        this.remoteState = {
+            disabled: false
+        }
+        this.element = React.createRef();
+    }
+
+    componentDidMount() {
+        this.initialize();
+    }
+
+    componentDidUpdate() {
+        if (this.state.field === null) {
+            this.initialize();
+        } else {
+            this.updateRemoteState();
+        }
+    }
+
+    initialize() {
+        if (!this.props.microform || this.state.field !== null) {
+            return;
+        }
+        let field = this.props.microform.createField(this.props.fieldType, {
+            disabled: this.props.disabled,
+        });
+        this.remoteState.disabled = this.props.disabled;
+        field.load(this.element.current);
+        if (this.props.onChange) {
+            field.on('change', this.props.onChange);
+        }
+        this.setState({ field });
+    }
+
+    updateRemoteState() {
+        if (this.props.disabled !== this.remoteState.disabled) {
```

Comment:
fedx consultant: What's the better way to trigger code when a prop changes?

Reply:
note that when you call `this.state.field.update()` it removes focus from the field, so calling it on every keystroke was preventing me from typing continuously into the field. I had to figure out some way to only call that function when this particular prop changed.

Reply:
`componentDidUpdate(prevProps, prevState)` - that version of `componentDidUpdate` will let you see past props.  Then you can compare them to the current `this.props` to determine if you should trigger code.

You can also do this by memoizing the function containing the code you want to trigger, if you want to get fancy.  But there's nothing wrong with the above and it's pretty understandable.

Memoization example for the curious: https://github.com/edx/frontend-app-learning/blob/master/src/courseware/CoursewareContainer.jsx#L72 (Note, `lodash.memoize` does NOT work the way you'd expect!)

Reply:
(On second thought, ignore all this memoization stuff - that first bit will do _just fine_)